app/scripts/app.js

```diff
@@ -485,6 +485,80 @@ angular
       LabelFilter.readPersistedState();
     });
   })
+  .run(function($location, $uibModal, AuthService, Constants) {
+    if (Math.sign(Constants.INACTIVITY_TIMEOUT_PERIOD) < 1 || $location.path() === '/logout' || $location.path() === '/error') {
+      return false;
+    }
+    var isLocalLogoutModalShown = false;
+    var updateLastInteraction = function() {
+      localStorage.setItem('origin-web-console-last-interaction-timestamp', new Date());
+    };
+    var updateLogOut = function(bool) {
+      localStorage.setItem('origin-web-console-inactivity-logout', bool);
+    };
+    var updateCloseAllModals = function(bool) {
+      localStorage.setItem('origin-web-console-dismiss-logout-modals', bool);
```

Comment:
i think we can eliminate this local storage variable.  when the user clicks the button to dismiss the dialogs and remain active that is going to set the last interaction time.

you can then have your modal add a listener on local storage https://stackoverflow.com/questions/28230845/communication-between-tabs-or-windows

that listener can dismiss the modal if it sees the key for the last interaction time is updated

you probably want to make sure that clicking the logout link does not reset the last interaction time, stopping propagation of the event from the ng-click of the button would probably fix this.